### PR TITLE
Fixes to example tools

### PR DESCRIPTION
**Pull request by wallyqs**
Apply similar fix as https://github.com/nats-io/nats.go/pull/456 and call exit 1 after showing the tool usage/help message.
Signed-off-by: Waldemar Quevedo <wally@synadia.com>

### CODE REVIEW

**Comment by wallyqs** [comment]
I think we should exit 0 at least in the case for when only showing help, then exit 1 on incorrect usage. I could add an `showUsageAndDie` function for that for example, then exit 0 when only showing help.

**Comment by derekcollison** [comment]
ok I agree.

**Comment by wallyqs** [comment]
Applied updates to the tools so that: 

- `-h`  is exit 0
- wrong usage, is exit 1
- wrong/unknown flags and arguments is exit 2

File: examples/nats-bench/main.go [hunk 1]
```diff
@@ -37,6 +37,10 @@ const (
 func usage() {
 	log.Printf("Usage: nats-bench [-s server (%s)] [--tls] [-np NUM_PUBLISHERS] [-ns NUM_SUBSCRIBERS] [-n NUM_MSGS] [-ms MESSAGE_SIZE] [-csv csvfile] <subject>\n", nats.DefaultURL)
 	flag.PrintDefaults()
+}
+
+func showUsageAndDie() {
```
Comment (derekcollison) [inline examples/nats-bench/main.go]: Maybe `ShowUsageAndExit(exitcode int)`?

Comment (wallyqs) [reply examples/nats-bench/main.go]: sounds good to me,  just updated

**Pull request merged**